Docker Image: humble_lint, need to install other dds.

# Need to install other dds

Docker Image: humble_lint doesn't have other dds(such as `connextdds, cyclonedds` `iceoryx`, etc) installed, which would cause run timer error when set env as other dds, for example:

```bash
 export RMW_IMPLEMENTATION=rmw_connextdds
```

## Description:

```bash
[ERROR] [1693811491.110734726] [rcl]: Error getting RMW implementation identifier / RMW implementation not installed (expected identifier of 'rmw_connextdds'), with error message 'failed to load shared library 'librmw_connextdds.so' due to dlopen error: librmw_connextdds.so: cannot open shared object file: No such file or directory, at /root/ros2_humble/src/ros2/rcutils/src/shared_library.c:99, at /root/ros2_humble/src/ros2/rmw_implementation/rmw_implementation/src/functions.cpp:65', exiting with 1., at /root/ros2_humble/src/ros2/rcl/rcl/src/rcl/rmw_implementation_identifier_check.c:139
```

```bash
[ERROR] [1693811534.722311830] [rcl]: Error getting RMW implementation identifier / RMW implementation not installed (expected identifier of 'rmw_connextdds'), with error message 'failed to load shared library 'librmw_connextdds.so' due to dlopen error: librmw_connextdds.so: cannot open shared object file: No such file or directory, at /root/ros2_humble/src/ros2/rcutils/src/shared_library.c:99, at /root/ros2_humble/src/ros2/rmw_implementation/rmw_implementation/src/functions.cpp:65', exiting with 1., at /root/ros2_humble/src/ros2/rcl/rcl/src/rcl/rmw_implementation_identifier_check.c:139
```

--- a/README.md
+++ b/README.md
@@ -50,6 +50,12 @@ example:
 sudo docker run --shm-size=1024m --name ros2_humble_lint --privileged -it -d -p 2337:22 -p 4907:4000 -v /Users/zhangtonghe/share:/share   tonghezhang/ros2_x86:humble_lint   /bin/bash
 ```
 
+### Limitation
+
+0. The Docker Image is based on `Ubuntu 20.04` and `ROS2 Humble`. It's not possible to run the image on a host with a different OS or ROS2 version.
+
+1. Using `iceoryx` as the RMW implementation is not always supported due to [#1776](https://github.com/eclipse-iceoryx/iceoryx/issues/1176), so we recommand `rmw_fastrtps_cpp` as the default RMW implementation. But you can find a solution in [#1776](https://github.com/eclipse-iceoryx/iceoryx/issues/1176) to use `iceoryx` as the RMW implementation.
+
 ### Docker update record
 
 [Update Record](doc/UpdateRecord.md)

--- a/doc/UpdateRecord.md
+++ b/doc/UpdateRecord.md
@@ -2,9 +2,29 @@
 
 ## 2023-09-03
 
-### Update
+### Update230903
 
 1. install clang
 2. add `colcon build --packages-up-to rclc`, we have rclc installed now.
 
 docker image: `tonghezhang/ros2_x86:humble_lint_0.1`
+
+## 2023-09-04
+
+### Update230904
+
+1. install rmw_cyclonedds_cpp and rmw_iceoryx_cpp
+2. add `colcon build --packages-up-to rmw_cyclonedds_cpp --packages-up-to rmw_iceoryx_cpp`, we have rmw_cyclonedds_cpp and rmw_iceoryx_cpp installed now.
+3. add `demo_nodes_cpp` for testing.
+
+docker image: `tonghezhang/ros2_x86:humble_lint_0.2`
+
+#### How to fix the problem of rmw_iceoryx_cpp
+
+Due to issue [#1776](https://github.com/eclipse-iceoryx/iceoryx/issues/1176)
+
+> To solve this problem locally yourself you could try to copy the file `iceoryx_hoofs/platform/win/include/iceoryx_hoofs/platform/acl.hpp` to
+> `iceoryx_hoofs/platform/linux/include/iceoryx_hoofs/platform/acl.hpp`.
+>
+> In Windows we do not support ACLs at all and replaced the libacl implementation with a stub implementation.
+> When you copy the windows acl.hpp file to the linux platform then linux would as well use the stub implementation of windows. It is like turning of ACLs.

